### PR TITLE
test: procedures in stress tool and performance improvements

### PR DESCRIPTION
**Pull request by jchappelow**
Based on top of https://github.com/kwilteam/kwil-db/pull/840

This started with the goal of simply adding procedures to the stress tool. and it now contains a good deal of related refactoring to increase transaction throughput.

With this PR and https://github.com/kwilteam/kwil-db/pull/840, we are finally able to fill blocks with small transaction <=1kB.

Needs a little cleanup since I was focusing on metrics and https://github.com/kwilteam/kwil-db/pull/840 while hacking on this, but for the most part it is what I wanted it to do.  Future work should wire up the foreign procedures schema in place of the simple one.

In addition to the stress tool updates, this also:
- performs more strict type assertions in acceptance tests (no need for conv package that permits too much)
- silence (via debug log instead of warn) a number of notices in the pg package that pertain to user dataset query errors

### CODE REVIEW

File: test/stress/foreign_social.kf [hunk 1]
```diff
@@ -0,0 +1,139 @@
+database foreign_social_media;
```
Comment (jchappelow) [inline test/stress/foreign_social.kf:1]: The foreign proc schemas aren't wired up yet, but I want to soon.

File: test/stress/hammer.go [hunk 2]
```diff
@@ -227,90 +259,62 @@ func hammer(ctx context.Context) error {
...
+	if maxPosters > 0 {
+		maxProcPost := maxPosters / 2
+		operation := "create post (proc)"
+		posters := make(chan struct{}, maxProcPost)
+		wg.Add(1)
+		go runLooped(ctx,
+			asyncFn(ctx, posters, h.printf, operation,
+				func() (<-chan asyncResp, error) {
+					content := bigData[:contentLen]
+					// content := bigData[:rand.Intn(maxContentLen)+1] // random.String(rand.Intn(maxContentLen) + 1) // randomBytes(maxContentLen)
```
Comment (charithabandi) [inline test/stress/hammer.go:309]: Maybe add a flag to configure we want fixed size payload or variable?

File: cmd/kwild/config/default_config.toml [hunk 3]
```diff
@@ -251,7 +251,7 @@ max_txs_bytes = 536870912
 max_tx_bytes = 4194304
 
 # Size of the cache (used to filter transactions we saw earlier) in transactions
-cache_size = 10000
+cache_size = 20000
```
Comment (charithabandi) [inline cmd/kwild/config/default_config.toml:254]: I think we should have higher cache_size here, atleast to the size of mempool. This is what keeps track of previously seen transactions and avoids rechecks and duplicate txs in the mempool.

Comment (jchappelow) [reply cmd/kwild/config/default_config.toml:254]: I would hope that a cache **miss** would just lead to a tx index check.  I'll increase this number, but do you have a link for how their mempool cache is works if there's a miss?  It's a badly cache-supported system that would result in duplicate txns in mempool in the event of a cache, so the purpose of this value doesn't seem clear.

File: test/stress/hammer.go (hunk 2 above)
Comment (jchappelow) [reply test/stress/hammer.go:309]: Yeah, good call.

File: cmd/kwild/config/default_config.toml (hunk 3 above)
Comment (jchappelow) [reply cmd/kwild/config/default_config.toml]: Bumped to 60,000 in https://github.com/kwilteam/kwil-db/pull/841/commits/a9ec473094c75cb8629e3e6f1a1830f51ce7aec9.
Will look into the functionality of their internal cache, but I think it's safer higher as you suggested.

File: test/stress/hammer.go (hunk 2 above)
Comment (jchappelow) [reply test/stress/hammer.go]: Added `-vl` flag for variable length: https://github.com/kwilteam/kwil-db/pull/841/commits/ebf62833893e8d6b23aed85e4e41b43c6f628447

**Comment by jchappelow** [comment]
Needs merge **without squash**.  A few independent changes might be cherry-picked for 0.8

File: cmd/kwild/config/default_config.toml (hunk 3 above)
Comment (charithabandi) [reply cmd/kwild/config/default_config.toml]: Yeah, else we might get into the hash recalculations, which we really wanna avoid as much as possible.

File: core/utils/json/json.go [hunk 4]
```diff
@@ -59,6 +59,10 @@ func convertJsonNumbers(val any) any {
 			return val.String()
 		}
 		return i
+	case string:
+		return val
+	case int64:
+		return val
```
Comment (jchappelow) [inline core/utils/json/json.go:65]: Worth mentioning that these are just optimizations.  These types would end up in default and go through reflection to check sliceiness  only to return as-is.

**Pull request merged**